docs/build/js-client/5-register-services.md

```diff
@@ -0,0 +1,166 @@
+# Register Aqua services (as plugins?)
+
+# TODO: adapt to JS Client
+
+# TODO: mention aqua arrays and their types convertion
+
+## Bullets
+
+- write aqua
+- export service
+- compile
+- register service
+
+## Understanding the compiler output
+
+```aqua
+service ServiceName:
+    -- service interface
+```
+
+For every exported `service` declaration the compiler will generate two entities: service interface under the name `{serviceName}Def` and a function named `register{serviceName}` with several overloads. First let's describe the most complete one using the following example:
+
+```typescript
+export interface ServiceNameDef {
+  //... service function definitions
+}
+
+export function registerServiceName(
+  peer: FluencePeer,
+  serviceId: string,
+  service: ServiceNameDef
+): void;
+```
+
+- `peer` - the Fluence Peer instance where the handler should be registered. The peer can be omitted. In that case the default Fluence Peer will be used instead
+- `serviceId` - the name of the service id. If the service was defined with the default service id in aqua code, this argument can be omitted.
+- `service` - the handler for the service.
+
+Depending on whether or not the services was defined with the default id the number of overloads will be different. In the case it **is defined**, there would be four overloads:
+
+```typescript
+// (1)
+export function registerServiceName(
+  //
+  service: ServiceNameDef
+): void;
+
+// (2)
+export function registerServiceName(
+  serviceId: string,
+  service: ServiceNameDef
+): void;
+
+// (3)
+export function registerServiceName(
+  peer: FluencePeer,
+  service: ServiceNameDef
+): void;
+
+// (4)
+export function registerServiceName(
+  peer: FluencePeer,
+  serviceId: string,
+  service: ServiceNameDef
+): void;
+```
+
+1. Uses default Fluence Peer and the default id taken from aqua definition
+2. Uses default Fluence Peer and specifies the service id explicitly
+3. The default id is taken from aqua definition. The peer is specified explicitly
+4. Specifying both peer and the service id.
+
+If the default id **is not defined** in aqua code the overloads will exclude ones without service id:
+
+```typescript
+// (1)
+export function registerServiceName(
+  serviceId: string,
+  service: ServiceNameDef
+): void;
+
+// (2)
+export function registerServiceName(
+  peer: FluencePeer,
+  serviceId: string,
+  service: ServiceNameDef
+): void;
+```
+
+1. Uses default Fluence Peer and specifies the service id explicitly
+2. Specifying both peer and the service id.
+
+### Type conversion
+
+Same as in call function. But no arrow types
+
+### Service interface
+
+The service interface type follows closely the definition in aqua code. It has the form of the object which keys correspond to the names of service members and the values are functions of the type translated from aqua definition (see [Type conversion](#type-conversion)). For example, for the following aqua definition:
```

Comment:
It is unclear whether "For example..." and "The typescript interface..." below make one sentence or not. If not, I would suggest to state explicitly what happens in "For example..."

Reply:
Thanks for you comment. I think we have to postpone these section to after the Devner event